data/data/aws/main.tf

```diff
@@ -106,6 +94,10 @@ resource "aws_route53_record" "etcd_cluster" {
   records = ["${formatlist("0 10 2380 %s", aws_route53_record.etcd_a_nodes.*.fqdn)}"]
 }
 
+data "aws_route53_zone" "public" {
```

Comment:
What is the motivation behind bringing this here from the route53 module? I would have expected that we would want to go the other way with these route53 resources and bring `aws_route53_record.etcd_a_nodes`, `aws_route53_record.etcd_cluster`, and `aws_route53_zone.int` to the route53 module instead of bringing route53 resources out of the route53 module.

Reply:
with how the current `r53` module is setup it takes private zone id and base_domain, and I think pulling the public zone out makes it so that the module takes ids for both for better consistency.

from https://github.com/openshift/installer/pull/1280/commits/d6de59aa39811ee0511998a8d251837026ac8ef5
```
Also moves the public zone discover to root module and pass the `public_route53_id`, similar to private zone, to `route53` zone.
```

and for moving the etcd records into the module, that makes sense. let me see and if the change is simple i'll do that.

Reply:
Well, my contention is that `aws_route32_zone.int` would be better placed in the route53 module. If that were moved, then there would be consistency in that neither public zone id nor private zone id would be passed into the route53 module.

Reply:
How would you then turn off either of them (not that you can do them right now..)

Reply:
I don't know. How would you turn off either of them if they were in main instead of the route53 module? How would that be different if they were in the route53 module?

Reply:
for example if we move public and private zone into module, then the module needs 2 variables... `base_domain` `cluster_domain`. And if I want to switch off the public zone records, I will have to pass empty `base_domain` IMO i don't like this indicator.

on the other hand if we pass zone ids, in the root module i pass empty zone id for public zone and that IMO is better indicator.

thoughts?

Reply:
If we were to make the public zone records optional, then there would need to be a variable coming into terraform to cover that configuration. That same variable would be passed down to the route53 module. I would not suggest overloading the semantics of `base_domain` to use it to indicate whether the public zone records are off or on. In my opinion, adding an additional variable to the route53 module that explicitly configures whether the public zone records are enabled is a worthwhile trade-off for keeping route53-specific resources contained within the route53 module.

Reply:
done with https://github.com/openshift/installer/pull/1280/commits/d4ce44e391bdcef7c476c887f6234ee741b888d7